Source/Diagnostics/ReducedDiags/FieldProbe.cpp

```diff
@@ -0,0 +1,250 @@
+        constexpr int index_Ex = 0;
+        constexpr int index_Ey = 1;
+        constexpr int index_Ez = 2;
+        constexpr int index_absE = 3;
+        constexpr int index_Bx = 4;
+        constexpr int index_By = 5;
+        constexpr int index_Bz = 6;
+        constexpr int index_absB = 7;
+
+        // Prepare interpolation of field components to cell center
+        // The arrays below store the index type (staggering) of each MultiFab, with the third
+        // component set to zero in the two-dimensional case.
+        amrex::IndexType const Extype = Ex.ixType();
+        amrex::IndexType const Eytype = Ey.ixType();
+        amrex::IndexType const Eztype = Ez.ixType();
+        amrex::IndexType const Bxtype = Bx.ixType();
+        amrex::IndexType const Bytype = By.ixType();
+        amrex::IndexType const Bztype = Bz.ixType();
+
+        amrex::Real hv_Ex, hv_Ey, hv_Ez, hv_E, hv_Bx, hv_By, hv_Bz, hv_B;
+        int probe_proc = -1;
+        // MFIter loop to interpolate fields to cell center and get maximum values
+#ifdef AMREX_USE_OMP
+#pragma omp parallel if (amrex::Gpu::notInLaunchRegion())
+#endif
+        for (amrex::MFIter mfi(Ex, TilingIfNotGPU()); mfi.isValid(); ++mfi ) {
+            const amrex::Box& box = mfi.validbox();
+
+            if(box.contains(i_probe, j_probe, k_probe)) {
+                // Make the box cell centered in preparation for the interpolation (and to avoid
+                // including ghost cells in the calculation)
+                const auto &arrEx = Ex[mfi].array();
+                const auto &arrEy = Ey[mfi].array();
+                const auto &arrEz = Ez[mfi].array();
+                const auto &arrBx = Bx[mfi].array();
+                const auto &arrBy = By[mfi].array();
+                const auto &arrBz = Bz[mfi].array();
+
+                amrex::ParticleReal Ex_interp = 0._rt, Ey_interp = 0._rt, Ez_interp = 0._rt;
+                amrex::ParticleReal Bx_interp = 0._rt, By_interp = 0._rt, Bz_interp = 0._rt;
+
+                amrex::Array<amrex::Real,3> v_galilean = {{0}};
+                const auto& xyzmin = WarpX::GetInstance().LowerCornerWithGalilean(box,v_galilean,
+                                                                                  lev);
+                const std::array<Real,3>& dx = WarpX::CellSize(lev);
+                amrex::GpuArray<amrex::Real, 3> dx_arr = {dx[0], dx[1], dx[2]};
+                amrex::GpuArray<amrex::Real, 3> xyzmin_arr = {xyzmin[0], xyzmin[1], xyzmin[2]};
+
+                //Interpolating to the probe point
+                doGatherShapeN (x_probe, y_probe, z_probe, Ex_interp, Ey_interp, Ez_interp,
+                                Bx_interp, By_interp, Bz_interp, arrEx, arrEy, arrEz, arrBx,
+                                arrBy, arrBz, Extype, Eytype, Eztype, Bxtype, Bytype, Bztype,
+                                dx_arr, xyzmin_arr, amrex::lbound(box),
+                                WarpX::n_rz_azimuthal_modes, WarpX::nox, false);
```

Comment:
You don't necessarily have to use the same interpolation order as for the field gathering `WarpX::nox` but it's fine as well to use it. In any case, it could be good to specify what you choose to do in the parameters doc.

Reply:
Good point. To be as flexible as possible I added an optional parameter to specify the interpolation order, with 1 as default value.